docs/source/topic/allowing.md

```diff
@@ -0,0 +1,143 @@
+Any user that doesn't meet any of the allow rules will not be allowed.
+
+| user  | allowed | reason                                                  |
+| ----- | ------- | ------------------------------------------------------- |
+| art   | True    | in `allowed_users`                                      |
+| amena | True    | member of `preservation`                                |
+| tlacy | False   | not in `allowed_users` and not member of `preservation` |
+
+### `allow_all`
+
+The first and simplest way to allow access is to any user who can successfully authenticate:
+
+```python
+c.OAuthenticator.allow_all = True
+```
+
+This is appropriate when you use an authentication provider (e.g. an institutional single-sign-on provider), where everyone who has an account in the provider should have access to your Hub.
+It may also be appropriate for unadvertised short-lived hubs, e.g. dedicated hubs for workshops that will be shutdown after a day, where you may decide it is acceptable to allow anyone who finds your hub to login.
+
+If `allow_all` is enabled, no other `allow` configuration will have any effect.
+
+```{seealso}
+Configuration documentation for {attr}`.OAuthenticator.allow_all`
+```
+
+### `allowed_users`
+
+This is top-level JupyterHub configuration, shared by all Authenticators.
+This specifies a list of users that are allowed by name.
+This is the simplest authorization mechanism when you have a small group of users whose usernames you know:
+
+```python
+c.OAuthenticator.allowed_users = {"mensah", "ratthi"}
+```
+
+If this is your only configuration, only these users will be allowed, no others.
+
+Note that any additional usernames in the deprecated `admin_users` configuration will also be added to the `allowed_users` set.
+
+```{seealso}
+Configuration documentation for {attr}`.OAuthenticator.allowed_users`
+```
+
+### `allow_existing_users`
+
+JupyterHub can allow you to add and remove users while the Hub is running via the admin page.
+If you add or remove users this way, they will be added to the JupyterHub database, but their ability to login will not be affected unless they are also granted access via an `allow` rule.
+
+To enable managing users via the admin panel, set
+
+```python
+c.OAuthenticator.allow_existing_users = True
+```
+
+```{warning}
+Enabling `allow_existing_users` means that _removing_ users from any explicit allow mechanisms will no longer revoke their access.
+Once the user has been added to the database, the only way to revoke their access to the hub is to remove the user from JupyterHub entirely, via the admin page.
+```
+
+```{seealso}
+Configuration documentation for {attr}`.OAuthenticator.allow_existing_users`
+```
+
+### provider-specific rules
+
+Each OAuthenticator provider may have its own provider-specific rules to allow groups of users access, such as:
+
+- {attr}`.CILogonOAuthenticator.allowed_idps`
```

Comment:
This list is not comprehensive. Should it be? I'm not sure! It will certainly get out-of-date if it's supposed to be.

Reply:
I think it's fine for this to be a non-comprehensive set of examples.
We've got provider specific pages e.g. https://oauthenticator.readthedocs.io/en/latest/tutorials/provider-specific-setup/providers/gitlab.html
In future I think it'd be nice to put more explanations on those pages, so instead of just listing the additional config properties we could add some text to tie it in with the explanation on this page.

Reply:
Can you use another example than Cilogons allowed_idps? It isn't really an allow config like the others, but a requirement to be from an idp listed.

See https://github.com/jupyterhub/oauthenticator/issues/683 and https://github.com/jupyterhub/oauthenticator/pull/685

Reply:
removed `allowed_idps`